stake-pool/program/src/state.rs

```diff
@@ -67,6 +67,8 @@ pub enum State {
 }
 
 impl State {
+    /// Length of state data when serialized
+    pub const LEN: usize = size_of::<u8>() + size_of::<StakePool>();
     /// Deserializes a byte buffer into a [State](struct.State.html).
     /// TODO efficient unpacking here
     pub fn deserialize(input: &[u8]) -> Result<State, ProgramError> {
```

Comment:
I can't comment further down, but the deserialization logic is probably still broken on line 81, since `unpack` is still receiving the entire input, and not `input[1..]`

Reply:
Right, this was here before me, will fix it.